src/Symfony/Bridge/Doctrine/PropertyInfo/DoctrineExtractor.php

```diff
@@ -110,10 +110,20 @@ public function getTypes($class, $property, array $context = [])
                 $associationMapping = $metadata->getAssociationMapping($property);
 
                 if (isset($associationMapping['indexBy'])) {
-                    $indexColumn = $associationMapping['indexBy'];
                     /** @var ClassMetadataInfo $subMetadata */
                     $subMetadata = $this->classMetadataFactory->getMetadataFor($associationMapping['targetEntity']);
-                    $typeOfField = $subMetadata->getTypeOfField($subMetadata->getFieldForColumn($indexColumn));
+                    $fieldName = $subMetadata->getFieldForColumn($associationMapping['indexBy']);
+                    $typeOfField = $subMetadata->getTypeOfField($fieldName);
+
+                    // Maybe $fieldName is an association
+                    if (null === $typeOfField) {
+                        $associationMapping = $subMetadata->getAssociationMapping($fieldName);
+
+                        /** @var ClassMetadataInfo $subMetadata */
+                        $indexProperty = $subMetadata->getSingleAssociationReferencedJoinColumnName($fieldName);
```

Comment:
Shouldn't we add some kind of check that the column name used here is in fact the join column?